cromshell

```diff
@@ -427,6 +456,42 @@ function populateLastWorkflowId()
   WORKFLOW_ID=$( tail -n1 ${CROMWELL_SUBMISSIONS_FILE} | awk '{print $3}' )
 }
 
+function alias_workflow() 
+{
+	local alias_name=${2}
+
+	# Check that we have an alias and workflow defined:
+	if [[ ${#alias_name} -eq 0 ]] || [[ ${#2} -eq 0 ]] ; then
+		error "ERROR: Cannot process alias name."
+		error "       Please specify both a workflow ID and an alias."
+		return 1
+	fi
+
+	# Check if the alias already exists:
+	awk '{print $6}' ${CROMWELL_SUBMISSIONS_FILE} | grep -v '^[ \t]*$' | grep -q "^${alias_name}$"
+	local r=$?
+	if [[ $r -eq 0 ]] ; then
+		error "ERROR: Alias already exists: ${alias_name}"
+		error "       Please choose another alias."
+		return 1
+	fi
+
+	# Get the workflow ID:
+	populateWorkflowIdAndServerUrl ${1}
+
+	error "Aliasing ${WORKFLOW_ID} -> ${alias_name}"
+	cp ${CROMWELL_SUBMISSIONS_FILE} ${CROMWELL_SUBMISSIONS_FILE}.bak
+	awk "BEGIN { FS=OFS=\"\t\" } { if (\$3 == \"${WORKFLOW_ID}\") { \$6=\"${alias_name}\"; print} else {print;} }" ${CROMWELL_SUBMISSIONS_FILE}.bak > ${CROMWELL_SUBMISSIONS_FILE} 
```

Comment:
I don't really awk so I'm not sure I can review this line, but I assume it's putting the alias name in the matching workflow id line?

Reply:
Indeed it is.

Quick rundown.

1) Set input and output field separators to tab.
2) if the third field is our workflow ID, set the sixth field to our alias and print the line. 
    otherwise print the line as is.

We use the newly created backup file as input and send updated our output to the original file.

Reply:
👍